src/python/pants/testutil/BUILD

```diff
@@ -13,7 +14,9 @@ python_distribution(
 )
 
 python_sources(
-    sources=["*.py", "!rule_runner.py", "!test_base.py", "!pants_integration_test.py", "!_test.py"]
+    sources=["*.py", "!rule_runner.py", "!test_base.py", "!pants_integration_test.py", "!_test.py"],
+    # Match ICs with the python tag applied to the distribution wheel above.
+    interpreter_constraints=[">=3.7,<3.10"],
```

Comment:
This is confusing without more context since this is the default for the repo and shouldn't need to be repeated - that's just a quirk. How about improving things in 2 ways, by making the fix local and somewhat non-redundant by specifying `python_requires="">=3.7,<3.10"` in the `setup_py` above instead?

Either way though, the important comment still remains to be added, which is that we need this since we artificially restrict the IC to a minor version pin (3.7.*, etc) in CI to build platform specific wheels for each pin separately.

Reply:
> That seems a little funky, but maybe works?

`Requires-Python: <3.10,==3.8.*,>=3.7` is valid, but reduces to `Requires-Python: ==3.8.*` which is not what you want IIUC and that's why you had to fix.

Reply:
I think your fix wasn't a fix IOW and you need to use the `setup_py` means of specifying that I suggested. That will totally ignore the funky IC CI uses in its 3 passes.

Reply:
I didn't know you could set `interpreter_constraints` on the `setup_py` .. ? Will have to try that, can't find how that is picked up by reading the source code.

Reply:
You wrote the code in this PR to handle already-set `python_requires`!

Reply:
Mm, no. That didn't seem to work. I'll add that comment however, that makes sense.
And see if I can't find a way to get a more accurate `Requires-Python` metadata.

Reply:
It's incredibly strange that didn't work. The dist should definitely have the right metadata with that `setup_py` amendment. That begs the question though of what about the main `pantsbuild.pants` metadata? Metadata for a distribution should be uniform across all its artifacts across a given version. I bet the 3.7, 3.8 and 3.9 wheels now all violate that with different `Requires-Python` metadata.

Reply:
Doh. Of course. Haha 🤦🏽

Reply:
Will just have to remove the restriction on it (right now it is only plugins that may set it) as it is on the restricted list of kwargs to use.

Reply:
> Metadata for a distribution should be uniform across all its artifacts across a given version.

FWIW - this is a dirty secret of how Poetry, PDM and Pex locking in universal mode can work at all. If these tools did not make that assumption, they'd be forced to download every wheel for a given project version to find out its unique metadata! That would be insanely slow; so all are technically incorrect in taking this shortcut; but the fact is they all do.

Reply:
OK, so the `setup_py` fix seems to have done the treat. I'll leave the CI to run for the night. Checking back in tomorrow or Monday. :)

Thanks for all the feedback and guidance.

Reply:
The `pantsbuild.pants` dist (c.f.: https://github.com/pantsbuild/pants/pull/15071#discussion_r846672293 above) was never correspondingly fixed. I let this slip and will fix presently.

Reply:
Oops, thanks.